### PR TITLE
Documented issue related before_destroy with dependent: :destroy

### PR DESCRIPTION
**Pull request by vishalzambre**
To address issues #3458, #20954, #670 It is good to have documented it at http://edgeguides.rubyonrails.org/active_record_callbacks.html

### CODE REVIEW

File: guides/source/active_record_callbacks.md [hunk 1]
```diff
@@ -117,6 +117,8 @@ Here is a list with all the available Active Record callbacks, listed in the sam
 
 WARNING. `after_save` runs both on create and update, but always _after_ the more specific callbacks `after_create` and `after_update`, no matter the order in which the macro calls were executed.
 
+NOTE: `before_destroy` should placed before `dependent: :destroy` associations or use `prepend: true` to execute it before `dependent: :destroy`
```
Comment (maclover7) [inline guides/source/active_record_callbacks.md]: `before_destroy` callback should be placed before `dependent: :destroy` associations (or use the `prepend: true` option), to ensure they execute before the records are deleted by `dependent: :destroy`.

**Review by maclover7 — CHANGES_REQUESTED** [review]
Please update the text per my comment. Thank you for the PR!

**Comment by vishalzambre** [comment]
@maclover7 Updated text as per comment.

**Comment by maclover7** [comment]
Thanks @vishalzambre! Can you squash down your commits to one, and add `[ci skip]` to the commit message? Looks ready to merge after that.

**Comment by vishalzambre** [comment]
@maclover7 squashed changes.

**Review by maclover7 — APPROVED** [review]

**Comment by maclover7** [comment]
:+1: Thank you for your contribution!

:heart: :green_heart: :blue_heart: :yellow_heart: :purple_heart:

**Comment by mstroming** [comment]
I see this documented in the Edge guide, but not here:

http://guides.rubyonrails.org/active_record_callbacks.html

What's the difference between those two guides? When people are Googling for this stuff, they're not automatically going to understand the difference.

**Pull request merged**